src/backend/cod4.c

```diff
@@ -3852,4 +3852,46 @@ code *cdcmpxchg(elem *e, regm_t *pretregs)
     return cat4(cr1,cr,cl,c);
 }
 
+/*************************
+ * Generate code for OPprefetch
+ */
+
+code *cdprefetch(elem *e, regm_t *pretregs)
+{
+    /* Generate the following based on e2:
+     *    0: prefetch0
+     *    1: prefetch1
+     *    2: prefetch2
+     *    3: prefetchnta
+     *    4: prefetchw
+     *    5: prefetchwt1
+     */
+    //printf("cdprefetch\n");
+    elem *e1 = e->E1;
+
+    assert(*pretregs == 0);
+    assert(e->E2->Eoper == OPconst);
+    unsigned op;
+    unsigned reg;
+    switch (e->E2->EV.Vuns)
+    {
+        case 0: op = PREFETCH; reg = 0; break;  // PREFETCH0
+        case 1: op = PREFETCH; reg = 1; break;  // PREFETCH1
+        case 2: op = PREFETCH; reg = 2; break;  // PREFETCH2
+        case 3: op = PREFETCH; reg = 3; break;  // PREFETCHNTA
+        case 4: op = 0x0F0D;   reg = 1; break;  // PREFETCHW
+        case 5: op = 0x0F0D;   reg = 2; break;  // PREFETCHWT1
+        default: assert(0);
+    }
+
+    freenode(e->E2);
+    code cs;
+    code *c = getlvalue(&cs,e1,0);
+    cs.Iop = op;
+    cs.Irm |= modregrm(0,reg,0);
```

Comment:
My knowledge is a bit rusted, how/where is the operand r/m specified when it's always 0 here.
Or does 0 mean EAX and getlvalue always uses that?